quantum/quantum.c

```diff
@@ -477,6 +477,13 @@ bool process_record_quantum(keyrecord_t *record) {
     return process_action_kb(record);
 }
 
+void set_single_default_layer(uint8_t default_layer) {
+#if defined(AUDIO_ENABLE) && defined(DEFAULT_LAYER_SONGS)
+    PLAY_SONG(default_layer_songs[default_layer]);
+#endif
+    default_layer_set((layer_state_t)1 << default_layer);
+}
+
 void set_single_persistent_default_layer(uint8_t default_layer) {
```

Comment:
Indeed. The reason why I didn't is that I did not want to mess with the order of operations here. I'm never sure how people are using it.

Reply:
@drashna I've applies this change

Reply:
Given the order of things, the main difference is when the audio plays (it is now after the eeprom is set, rather than before.   In both cases, the eeprom is set before the layer function is/was called, so that should be fine